src/util/persisted-data.test.ts

```diff
@@ -1,94 +1,106 @@
 /* eslint-env jest */
-import fs from 'fs-extra-promise';
-import jestFs from 'jest-plugin-fs';
-import yaml from 'js-yaml';
+import fs from 'fs-extra';
+
 import { isEqual } from 'lodash';
+import path from 'path';
 
 import { IRepo } from '../adapters/base';
 import { IMigrationContext } from '../migration-context';
 import { loadRepoList, updateRepoList } from './persisted-data';
 
-jest.mock('fs', () => require('jest-plugin-fs/mock')); // eslint-disable-line global-require
-
-const reposFixture = [{
-  owner: 'NerdWallet',
-  name: 'test',
-}];
 
-const makeContext = () => ({
+const makeContext = (workingDirectory: string) => ({
   migration: {
-    workingDirectory: '/migration',
+    workingDirectory,
   },
   adapter: {
     reposEqual: (r1: IRepo, r2: IRepo) => isEqual(r1, r2),
   },
 } as IMigrationContext);
 
 describe('persisted-data', () => {
-  beforeEach(() => jestFs.mock({
-    '/migration/repos.json': JSON.stringify(reposFixture),
-  }));
-  afterEach(() => jestFs.restore());
 
   it('loads repo list from a file', async () => {
-    const repos = await loadRepoList(makeContext());
-    expect(repos).toEqual([{ owner: 'NerdWallet', name: 'test' }]);
+    const workingDirectory = path.join(__dirname, '../../fixtures/artifacts/.shepherd/load-repo-from-file/');
+    const repos = await loadRepoList(makeContext(workingDirectory));
+    // fs.unlinke(path.join(workingDirectory,'/repos.json'));
+    expect(repos).toEqual([{'defaultBranch': 'master', 'name': 'shepherd', 'owner': 'NerdWalletOSS'}]);
   });
 
   it('returns null if the file does not exist', async () => {
-    await fs.unlinkAsync('/migration/repos.json');
-    const repos = await loadRepoList(makeContext());
+    const workingDirectory = path.join(__dirname, '../../fixtures/artifacts/.shepherd/no-repos-json/');
+    const repos = await loadRepoList(makeContext(workingDirectory));
     expect(repos).toEqual(null);
   });
 
   it('migrates from a YAML file to a JSON file', async () => {
-    const repos = await loadRepoList(makeContext());
-    expect(repos).toEqual([{ owner: 'NerdWallet', name: 'test' }]);
+    const workingDirectory = path.join(__dirname, '../../fixtures/artifacts/.shepherd/yaml-to-json/');
+    const source = path.join(workingDirectory,'repos-artifact.yml');
```

Comment:
Formatting nit: can you ensure there are spaces between function arguments?

We should add Prettier to this repo in another PR to avoid formatting comments like this 🙂

Reply:
Absolutely!  I was thinking of doing that along with yamllinter.

Reply:
Yes! Linting as a whole could use some love. ESLint is now the preferred way to lint TypeScript, so we should move to that from TSLint.

Reply:
Yay! ESlint ❤️